src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -583,10 +617,11 @@ private static unsafe int GetNextUnicodeScalarValueFromUtf16Surrogate(ref char*
 
         private static int HexToInt(char h)
         {
-            return (h >= '0' && h <= '9') ? h - '0' :
-            (h >= 'a' && h <= 'f') ? h - 'a' + 10 :
-            (h >= 'A' && h <= 'F') ? h - 'A' + 10 :
-            -1;
+            return IsIntBetween(h, '0', '9') ?
```

Comment:
I know it was like this before but: I find this sort of ternary statement with 3 possible outcomes confusing. Can we make it a series of if statements?